pkg/apis/rbac/v1alpha1/types.go

```diff
@@ -0,0 +1,284 @@
+// It adds who information via Users and Groups and namespace information by which namespace it exists in.  RoleBindings in a given
+// namespace only have effect in that namespace (excepting the master namespace which has power in all namespaces).
+type RoleBinding struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	v1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// UserNames holds all the usernames directly bound to the role
+	UserNames []string `json:"userNames" protobuf:"bytes,2,rep,name=userNames"`
+	// GroupNames holds all the groups directly bound to the role
+	GroupNames []string `json:"groupNames" protobuf:"bytes,3,rep,name=groupNames"`
+	// Subjects hold object references to authorize with this rule
+	Subjects []v1.ObjectReference `json:"subjects" protobuf:"bytes,4,rep,name=subjects"`
+
+	// RoleRef can only reference the current namespace and the global namespace
+	// If the RoleRef cannot be resolved, the Authorizer must return an error.
+	// Since Policy is a singleton, this is sufficient knowledge to locate a role
+	RoleRef v1.ObjectReference `json:"roleRef" protobuf:"bytes,5,opt,name=roleRef"`
+}
+
+// Policy is a object that holds all the Roles for a particular namespace.  There is at most
+// one Policy document per namespace.
+type Policy struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	v1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// LastModified is the last time that any part of the Policy was created, updated, or deleted
+	LastModified unversioned.Time `json:"lastModified" protobuf:"bytes,2,opt,name=lastModified"`
+
+	// Roles holds all the Roles held by this Policy, mapped by Role.Name
+	Roles []NamedRole `json:"roles" protobuf:"bytes,3,rep,name=roles"`
+}
+
+// PolicyBinding is a object that holds all the RoleBindings for a particular namespace.  There is
+// one PolicyBinding document per referenced Policy namespace
+type PolicyBinding struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	v1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// LastModified is the last time that any part of the PolicyBinding was created, updated, or deleted
+	LastModified unversioned.Time `json:"lastModified" protobuf:"bytes,2,opt,name=lastModified"`
+
+	// PolicyRef is a reference to the Policy that contains all the Roles that this PolicyBinding's RoleBindings may reference
+	PolicyRef v1.ObjectReference `json:"policyRef" protobuf:"bytes,3,opt,name=policyRef"`
+	// RoleBindings holds all the RoleBindings held by this PolicyBinding, mapped by RoleBinding.Name
+	RoleBindings []NamedRoleBinding `json:"roleBindings" protobuf:"bytes,4,rep,name=roleBindings"`
+}
+
+// NamedRole relates a Role with a name
+type NamedRole struct {
```

Comment:
Since we're considering dropping policyBinding that mapping would go away.

Forgive me if this is a naive question, but is there a reason not to put the name directly on the role?